bidict/_bidict.py

```diff
@@ -9,54 +9,60 @@ class bidict(BidirectionalMapping, MutableMapping):
     """Mutable bidirectional map type."""
 
     def _del(self, key):
-        val = self._fwd[key]
-        del self._fwd[key]
+        val = self._fwd.pop(key)
         del self._inv[val]
         return val
 
     def __delitem__(self, key):
-        """Like :py:meth:`dict.__delitem__`, keeping bidirectionality intact."""
+        """Like :py:meth:`dict.__delitem__`, maintaining bidirectionality."""
         self._del(key)
 
     def __setitem__(self, key, val):
         """
         Set the value for *key* to *val*.
 
+        If *key* is already associated with *val*, this is a no-op.
+
         If *key* is already associated with a different value,
-        the old value will be replaced with *val*.
-        Use :attr:`put` to raise an exception in this case instead.
+        the old value will be replaced with *val*,
+        as with :py:meth:`dict.__setitem__`.
 
         If *val* is already associated with a different key,
         an exception is raised
-        to protect against accidentally replacing the existing mapping.
-
-        If *key* is already associated with *val*, this is a no-op.
+        to protect against accidental removal of the key
+        that's currently associated with *val*.
 
-        Use :attr:`forceput` to unconditionally associate *key* with *val*,
-        replacing any existing mappings as necessary to preserve uniqueness.
+        Use :attr:`put` instead if you want to specify different behavior in
+        the case that the provided key or value collides with an existing one.
+        Or use :attr:`forceput` to unconditionally associate *key* with *val*,
+        replacing any existing key or value as necessary to preserve uniqueness.
 
-        :raises bidict.ValueExistsException: if attempting to set a mapping
-            with a non-unique value.
+        :raises bidict.ValueExistsException: if *val* is not unique.
         """
-        self._put(key, val, overwrite_key=False, overwrite_val=True)
+        self._put(key, val, OVERWRITE, RAISE)
 
-    def put(self, key, val):
+    def put(self, key, val, key_clbhv=RAISE, val_clbhv=RAISE):
```

Comment:
I don't think kwarg names this cryptic should be in the public API. More verbose, but on_key_collision/on_value_collision would be better.